update doc filters.md (#657)
megaease/easegress: doc/reference/filters.md

```yaml
kind: Proxy
name: proxy-example-1
pools:
- servers:
  - url: http://127.0.0.1:9095 
```

> Pool without `filter` is considered as the main pool, other pools with `filter` are considered as candidate pools. Proxy first checks if one of the candidate pools can process a request. For example, the first candidate pool in the below configuration selects and processes request with header `X-Candidate:candidate`, the second candidate pool randomly selects and processes 40% of requests, and the main pool processes the other 60% of requests. 

Reviewer: ```suggestion
Pool without `filter` is considered the main pool, other pools with `filter` are considered candidate pools. Proxy first checks if one of the candidate pools can process a request. For example, the first candidate pool in the below configuration selects and processes requests with the header `X-Candidate:candidate`, the second candidate pool randomly selects and processes 400‰ of requests, and the main pool processes the other 600‰ of requests. 
```

```yaml
kind: Proxy
name: proxy-example-4
pools:
- serverTags: ["v2"] 
  serviceName: service-001
  serviceRegistry: eureka-service-registry-example
  loadBalance:
    policy: roundRobin
```

### Configuration
| Name | Type | Description | Required |
| ---- | ---- | ----------- | -------- |

> | pools | [proxy.ServerPoolSpec](#proxyserverpoolspec) | The pool without `filter` is considered as `mainPool`, other pools with `filter` are considered as `candidatePools`. When `Proxy` get a request, it first goes through the pools in `candidatePools`, and if one of the pools' filter matches the request, servers of this pool handle the request, otherwise, the request is passed to `mainPool` | Yes |  

Reviewer: ```suggestion
| pools | [proxy.ServerPoolSpec](#proxyserverpoolspec) | The pool without `filter` is considered the main pool, other pools with `filter` are considered candidate pools, and a `Proxy` must contain exactly one main pool. When `Proxy` gets a request, it first goes through the candidate pools, and if one of the pool's filter matches the request, servers of this pool handle the request, otherwise, the request is passed to the main pool. | Yes |  
```

| mirrorPool | [proxy.ServerPoolSpec](#proxyserverpoolspec) | Define a mirror pool, requests are sent to this pool simultaneously when they are sent to candidate pools or main pool | No |
| compression | [proxy.CompressionSpec](#proxyCompressionSpec) | Response compression options | No |
| mtls | [proxy.MTLS](#proxymtls) | mTLS configuration | No |
| maxIdleConns | int | Controls the maximum number of idle (keep-alive) connections across all hosts. Default is 10240 | No |
| maxIdleConnsPerHost | int | Controls the maximum idle (keep-alive) connections to keep per-host. Default is 1024 | No |

> | serverMaxBodySize | int64 | Max size of request body. Default value if 4 * 1024 * 1024 | No |

Reviewer: ```suggestion
| serverMaxBodySize | int64 | Max size of response body. the default value is 4MB. Responses with a body larger than this option are discarded.  When this option is set to `-1`, Easegress takes the response body as a stream and the body can be any size, but some features are not possible in this case. | No |
```

| decompressFail | the request body can not be decompressed |
| compressFail   | the request body can not be compressed   |

## RequestBuilder

The RequestBuilder create new requests according to configuration. 

The example configuration below create a http request with method `GET`, url `http://127.0.0.1:8080`, headers `X-Mock-Header:mock-value` and body `this is body`.  

```yaml 
name: requestbuilder-example-1
kind: RequestBuilder
protocol: http
template: |
  method: get 
  url: http://127.0.0.1:8080
  headers:
    X-Mock-Header: 
    - mock-value
  body: "this is body" 
```

> Although `template` is string, it content should be following yaml format. For example:

Reviewer: ```suggestion
Although `template` is a string, its content should follow the YAML format. For example:
```
